File: Algorithms/src/1518. Water Bottles/Code/solution.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int total = numBottles;
        int empties = numBottles;

        while (empties >= numExchange) {
            int newFull = empties / numExchange;
            total += newFull;
            empties = newFull + (empties % numExchange);
        }

        return total;
    }
}

File: Algorithms/src/1518. Water Bottles/Code/solution.go
func numWaterBottles(numBottles int, numExchange int) int {
    total := numBottles
    empties := numBottles

    for empties >= numExchange {
        newFull := empties / numExchange
        total += newFull
        empties = newFull + empties%numExchange
    }

    return total
}
